File: src/i_introduction/_7_Nullable_Types/JavaCode7.java
package i_introduction._7_Nullable_Types;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import util.JavaCode;

public class JavaCode7 extends JavaCode {
    public void sendMessageToClient(@Nullable Client client, @Nullable String message, @NotNull Mailer mailer) {
        if (client == null || message == null) return;

        PersonalInfo personalInfo = client.getPersonalInfo();
        if (personalInfo == null) return;

        String email = personalInfo.getEmail();
        if (email == null) return;

        mailer.sendMessage(email, message);
    }
}

File: src/i_introduction/_7_Nullable_Types/JavaCode7.kt
package i_introduction._7_Nullable_Types

import util.JavaCode

class JavaCode7 : JavaCode() {
    fun sendMessageToClient(client: Client?, message: String?, mailer: Mailer) {
        if (client == null || message == null) return

        val personalInfo = client.personalInfo ?: return

        val email = personalInfo.email ?: return

        mailer.sendMessage(email, message)
    }
}
